packages/core/src/demo/echo.js

```diff
@@ -41,36 +41,62 @@ const messageEchoCache = {};
   const apiClient = new APIClient({store: engine, urls: backend});
   const account = new Account(apiClient);
 
-  account.on(PayloadBundleType.TEXT, async data => {
-    const {conversation: conversationId, from, content, id: messageId, messageTimer} = data;
+  const handleIncomingMessage = async messageData => {
+    const {conversation: conversationId, content, from, id: messageId, messageTimer = 0, type} = messageData;
+
     logger.log(
-      `Message "${messageId}" in "${conversationId}" from "${from}":`,
-      content.text,
+      `Receiving: "${type}" ("${messageId}") in "${conversationId}" from "${from}":`,
+      content,
       messageTimer ? `(ephemeral message, ${messageTimer} ms timeout)` : ''
     );
 
-    const confirmationPayload = account.service.conversation.createConfirmation(messageId);
-    await account.service.conversation.send(conversationId, confirmationPayload);
+    if (messageId) {
```

Comment:
In the long run a message ID should not be the trigger of for a confirmation. There are specific content messages that need to be confirmed. Call messages won't be for example.